doc/adding_new_sensors.rst

```diff
@@ -0,0 +1,141 @@
+.. py:currentmodule:: lsst.ts.ess.common
+
+.. _lsst.ts.ess.common.adding_new_sensors:
+
+==================
+Adding New Sensors
+==================
+
+The ts_ess_common and ts_ess_controller projects contain code for a wide variety of sensors, both in terms of functionality and how to conntect to them.
+Occasionally a new sensor needs to be added.
+This document describes how to do that.
+
+Add New Sensor Code For An Unsupported Sensor Type
+==================================================
+
+There are several cases:
+
+* The sensor serves data via TCP/IP, without needing a special library that is hard to install.
+
+  In this case an ESS instance running on Kubernetes can directly connect to the sensor.
+  To support this, write a subclass of `data_client.BaseDataClient` that handles your sensor.
+  One example is `data_client.Young32400WeatherStationDataClient`.
+  Note that the `sensor.BaseSensor` and `device.BaseDevice` are not used for this case.
+
+  If communication needs a special library that is easy to install, add it to the standard distribution.
+
+* The sensor serves data via USB, FTDI or some other protocol that is not visible from Kubernetes.
+
+  In this case we connect the sensor to a Power over Ethernet (PoE) Raspberry Pi as designed by Oliver Weicha.
+  The RPi then relays the data to the `data_client.RPiDataClient`.
+  See :ref:`connected-to-rpi` for details on how to handle this.
+
+* The sensor serves data via an RS-232 or RS-485 serial interface.
+
+  There are two options:
+
+  * If you power the sensor in the conventional way, attach a serial-to-ethernet adapter and see the TCP/IP case above.
+  * If you want to power the sensor through the serial connector (a custom technique),
+    see the USB/FTDI case above and use a Power over Ethernet (PoE) Raspberry Pi as designed by Oliver Weicha.
+
+  In theory you can use an RPi even if the sensor is powered conventionally, but we do not recommend this for new sensors.
```

Comment:
I don't think this is true. There will be new sensors of existing types that will be connected to RPis, like the sensors that will be installed on the inside of the MTDome.

Reply:
Good point. I'll say "new types of sensors".